### PR TITLE
rework setInitialPosition()

### PR DESCRIPTION
**Pull request by osheroff**
- only do the setInitialPosition() call in the producer.  
- mild refactoring

@vanchi-zendesk

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java [hunk 1]
```diff
@@ -93,12 +93,9 @@ public void run() throws Exception {
 			if ( !skipEvent(event)) {
 				producer.push(event);
 
-				// TODO:  this isn't quite right: we need to only stop on table map events,
-				// although it's unclear to me whether you can have two row events in a row.
-				// I think maybe you can.
+				// TODO:  this isn't quite right: we need to only stop on table map events
 
 				BinlogPosition p = eventBinlogPosition(event);
-				this.config.setInitialPosition(p);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: do we use p after this? Maybe we can remove eventBinlogPosition as well?

**Comment by vanchi-zendesk** [comment]
:+1:

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: yup fixed

**Comment by vanchi-zendesk** [comment]
:+1:

**Pull request merged**